models.js

```diff
@@ -24,12 +40,22 @@ debuglog = function () {
         }
         throw name + " is not a valid axis name!";
     };
-    
+
     // helper function
     var objToXPath = function(something) {
         return something.toXPath();
     };
-    
+
+    var objToHashtag = function (xpath_) {
+        return hashtagConfig.toHashtag(xpath_) || xpath_.toHashtag();
+    };
+
+    var objToHashtagWithCombine = function(self, combineFunc) {
+        return function () {
+            return hashtagConfig.toHashtag(self) || combineFunc(objToHashtag).bind(self)();
```

Comment:
General idea is to try to convert this entire xpath to a hashtag and if it's not possible to translate it, then try to translate it's children. Something like `/data/filtered[/data/question1 = '1']` could be  translated to `/data/filtered[#form/question1 = '1']`